django/db/models/fields/related_descriptors.py

```diff
@@ -956,32 +956,23 @@ def count(self):
             constrained_target = self.constrained_target
             return constrained_target.count() if constrained_target else super().count()
 
-        def add(self, *objs):
-            if not rel.through._meta.auto_created:
-                opts = self.through._meta
-                raise AttributeError(
-                    "Cannot use add() on a ManyToManyField which specifies an "
-                    "intermediary model. Use %s.%s's Manager instead." %
-                    (opts.app_label, opts.object_name)
-                )
+        def add(self, *objs, through_defaults=None):
```

Comment:
I noticed that we use keyword-only arguments for `def set(self, objs, *, clear=False, through_defaults=None):`

Do we want to so something similar for other methods:
- `def add(self, *objs, *, through_defaults=None)`:
- `def create(self, *, through_defaults=None, **kwargs):`
- `def get_or_create(self, *, through_defaults=None, **kwargs):`
- `def update_or_create(self, *, through_defaults=None, **kwargs):`

Reply:
Sure, I will push those change, except the suggestion for `add()` gives invalid syntax.

Reply:
Oh. Strange. Not sure why that doesn't work. Thanks.